lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -348,6 +376,18 @@ static OptionalParseResult customTypeParser(AsmParser &parser, StringRef name,
 
     return result = RefType::get(type, true), success();
   }
+  if (name.equals("string")) {
+    if (isConst) {
+      parser.emitError(parser.getNameLoc(), "strings cannot be const");
+      return failure();
+    }
+    auto type =
+        parser.getChecked<StringType>(parser.getContext(), Phase::Hardware);
```

Comment:
Great, thanks for using `getChecked`!

Reply:
Add a error test for `firrtl.string` (no phase specifier?)? (as-is this call always fails, right?)